### PR TITLE
Bugfix/fronthaul segfault

### LINKED ISSUE #1438: [BUG] Certification: slave reset soon after being operational
A lot of certification tests failed in the recent nightly runs.
By looking at [the sheet that holds the nightly tests results](https://docs.google.com/spreadsheets/d/1NtSMcRHDHsdjtqxIsVrJLSFzxUQ1c2dTJbrC6xGV0Wo), the bug was not yet present in the tests from `6/10/2020 9:33:08` (https://gitlab.com/prpl-foundation/prplMesh/-/jobs/588529913).

It was present in the tests from `6/11/2020 8:51:54`  (https://gitlab.com/prpl-foundation/prplMesh/-/jobs/590350508).

The logs always show multiple `slave_reset` soon after the slave is operational. From agent_wlan0.log:
```
DEBUG 06:38:38:946 <2002205584> son_slave_thread.cpp[2137] --> send ACTION_BACKHAUL_CLIENT_ASSOCIATED_NOTIFICATION for client 00:50:43:22:56:5f
DEBUG 06:38:38:948 <2002205584> son_slave_thread.cpp[2160] --> non-prlMesh, not adding ClientAssociationEvent VS TLV
DEBUG 06:38:39:231 <2002205584> son_slave_thread.cpp[219] --> monitor socket disconnected - slave_reset()
DEBUG 06:38:39:231 <2002205584> son_slave_thread.cpp[138] --> slave_reset() #1 - start
DEBUG 06:38:39:232 <2002205584> son_slave_thread.cpp[3654] --> removing backhaul_manager_socket
DEBUG 06:38:39:232 <2002205584> son_slave_thread.cpp[3665] --> removing platform_manager_socket
DEBUG 06:38:39:240 <2002205584> son_slave_thread.cpp[3672] --> hostap_services_off() - done
INFO 06:38:39:241 <2002205584> son_slave_thread.cpp[3683] --> fronthaul stop
DEBUG 06:38:39:242 <2002205584> son_slave_thread.cpp[163] --> goto STATE_INIT
DEBUG 06:38:39:244 <2002205584> son_slave_thread.cpp[168] --> slave_reset() #1 - done

```

### PR DESCRIPTION
**Pull request by rmelotte**
The previous parsing of SNR and ShortTermRSSIAverage was failing,
leading to a segmentation fault of beerocks_fronthaul.

Parse them as strings instead. Since they contain space-separated
values, split them later.

Fixes https://github.com/prplfoundation/prplMesh/issues/1438

### CODE REVIEW

File: common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp [hunk 1]
```diff
@@ -689,19 +689,19 @@ bool mon_wlan_hal_dwpal::update_stations_stats(const std::string &vap_iface_name
     size_t numOfValidArgs[10] = {0}, replyLen = strnlen(reply, HOSTAPD_TO_DWPAL_MSG_LENGTH);
     uint64_t BytesSent = 0, BytesReceived = 0, PacketsSent = 0, PacketsReceived = 0,
              LastDataDownlinkRate = 0, LastDataUplinkRate = 0, Active = 0;
-    char ShortTermRSSIAverage[32][HOSTAPD_TO_DWPAL_VALUE_STRING_LENGTH] = {'\0'};
-    char SNR[32][HOSTAPD_TO_DWPAL_VALUE_STRING_LENGTH]                  = {'\0'};
-    FieldsToParse fieldsToParse[]                                       = {
+    char ShortTermRSSIAverage[32] = {'\0'};
```
Comment (dob71) [inline common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:692]: It looks like the old code was expecting to handle up to 32 values, the new code will be limited at about 8 values:
"-77 -78 -55 -77 -78 -55 -44 -66\0" <- 32 bytes
is that intentional?

File: common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp [hunk 2]
```diff
@@ -738,18 +738,20 @@ bool mon_wlan_hal_dwpal::update_stations_stats(const std::string &vap_iface_name
         }
     }
 
-    //save avarage RSSI in watt
-    for (uint8_t i = 0; i < numOfValidArgs[5]; i++) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(ShortTermRSSIAverage[i])));
+    //save average RSSI in watt
```
Comment (dob71) [inline common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:741]: That comment from the old code is incorrect, it should be mW (milliwatts)

File: common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp [hunk 3]
```diff
@@ -738,18 +738,20 @@ bool mon_wlan_hal_dwpal::update_stations_stats(const std::string &vap_iface_name
...
         if (s_float > beerocks::RSSI_MIN) {
             sta_stats.rx_rssi_watt += std::pow(10, s_float / float(10));
             sta_stats.rx_rssi_watt_samples_cnt++;
         }
     }
 
-    //save avarage SNR in watt
-    for (uint8_t i = 0; i < numOfValidArgs[6]; i++) {
-        float s_float = float(beerocks::string_utils::stoi(std::string(SNR[i])));
+    //save average SNR in watt
```
Comment (dob71) [inline common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp:751]: This comment from the old code is also incorrect, it's a ratio (just a number), it was probably meant to say it's converted from logarithmic to dynamic scale.

File: common/beerocks/bwl/include/bwl/mon_wlan_hal_types.h [hunk 4]
```diff
@@ -64,29 +64,29 @@ struct SVapStats {
 };
 
 struct SStaStats {
-    float rx_rssi_watt;
-    uint8_t rx_rssi_watt_samples_cnt;
-    float rx_snr_watt;
-    uint8_t rx_snr_watt_samples_cnt;
+    float rx_rssi_watt = 0;
```
Comment (dob71) [inline common/beerocks/bwl/include/bwl/mon_wlan_hal_types.h]: Just like the comments the names of the vars from the old code are also a bit off, should probably be:
rx_rssi_mw/rx_rssi_mw_samples_cnt
and
rx_snr_ratio/rx_snr_ratio_samples_cnt

**Review by dob71 — APPROVED** [review]
IMO it needs a few changes, but if this fix is needed urgently those changes can wait

**Review by arnout — APPROVED** [review]
I don't really like this fix because I think it skirts around the problem: the dwpal parser is supposed to do exactly what we now do in C code.

However, I don't want to spend time debugging dwpal. Probably the problem is in fact in the way we call dwpal - the `sizeof(SStaStats)` argument is pretty suspicious since we're not actually passing in a SStaStats structure... But again, instead of trying to debug this, I would much rather replace this crazy parse function with the reliable parser in C++ that @morantr has created.

That said, we need this to be fixed yesterday, so let's merge it.

**Review by tomereli — APPROVED** [review]

**Review by vitalybu — APPROVED** [review]

**Pull request merged**